$FreeBSD$

Hack configure to eat autogenerated qt options, to fix a bashism (type -p),
to avoid using expr, and to do the main Makefile make(1) compatible.

--- configure.orig	Wed Dec 10 01:10:03 2003
+++ configure	Sat Jan 10 15:52:00 2004
@@ -27,32 +27,32 @@
 while [ $# -gt 0 ]; do
 	case "$1" in
 		--prefix=*)
-			PREFIX=`expr "${1}" : "--prefix=\(.*\)"`
+			PREFIX="${1#--prefix=}"
 			shift
 			;;
 
 		--bindir=*)
-			BINDIR=`expr "${1}" : "--bindir=\(.*\)"`
+			BINDIR="${1#--bindir=}"
 			shift
 			;;
 
 		--libdir=*)
-			LIBDIR=`expr "${1}" : "--libdir=\(.*\)"`
+			LIBDIR="${1#--libdir=}"
 			shift
 			;;
 
 		--qtdir=*)
-			QTDIR=`expr "${1}" : "--qtdir=\(.*\)"`
+			QTDIR="${1#--qtdir=}"
 			shift
 			;;
 
 		--with-zlib-inc=*)
-			QC_WITH_ZLIB_INC=`expr "${1}" : "--with-zlib-inc=\(.*\)"`
+			QC_WITH_ZLIB_INC="${1#--with-zlib-inc=}"
 			shift
 			;;
 
 		--with-zlib-lib=*)
-			QC_WITH_ZLIB_LIB=`expr "${1}" : "--with-zlib-lib=\(.*\)"`
+			QC_WITH_ZLIB_LIB="${1#--with-zlib-lib=}"
 			shift
 			;;
 
@@ -81,6 +81,8 @@
 			shift
 			;;
 		--help) show_usage; exit ;;
+		--with-*)
+			shift ;;
 		*) show_usage; exit ;;
 	esac
 done
@@ -136,7 +138,7 @@
 		echo Warning: qmake not in \$QTDIR/bin/qmake
 		echo trying to find it in \$PATH
 	fi
-	qm=`type -p qmake`
+	qm=`which qmake`
 	if [ -x "$qm" ]; then
 		if [ "$QC_DEBUG" = "Y" ]; then
 			echo qmake found in $qm
@@ -917,13 +919,6 @@
 	echo
 	exit 1;
 fi
-cat >Makefile.tmp <<EOT
-export QTDIR = $QTDIR
-EOT
-cat Makefile >> Makefile.tmp
-rm -f Makefile
-cp -f Makefile.tmp Makefile
-rm -f Makefile.tmp
 
 echo
 echo Good, your configure finished.  Now run \'make\'.
